api/migrations/20200709183253-loadMemDocs.js

```diff
@@ -0,0 +1,336 @@
+        errors += 1;
+      }
+    }
+    console.log(`Collections loaded from mem-admin!`);
+
+    // Now that we have all the docs, we need to create records for each one in NRPTI
+    // then, follow that up by pulling the doc from mem-admin, and pushing it up to nrpti's s3
+    const nrpti = await mClient.collection('nrpti');
+
+    for(const mine of publishedMines) {
+      // find the mine in nrpti by permit ID
+      console.log(`Processing mine ${mine.memPermitID}`);
+      const nrptiMine = await nrpti.findOne({ _schemaName: 'MineBCMI', permitNumber: mine.memPermitID })
+
+      if (nrptiMine) {
+        // Found a mine, now lets create a record and upload up the docs
+        for(const collection of mine.collectionData) {
+          console.log(`Processing collection ${collection.displayName}`);
+          const allDocs = collection.mainDocuments.concat(collection.otherDocuments);
+          const allNewDocs = [];
+          console.log(`Fetched ${allDocs.length} documents. Creating NRPTI records/flavours...`);
+          for(const collectionDoc of allDocs) {
+            if (collection.type && collection.type.length > 0 && !collection.isForEAO) {
+              try {
+                const newDoc = await createMineDocument(nrpti, nrptiMine, collection, collectionDoc);
+                if (newDoc) {
+                  allNewDocs.push(new ObjectID(newDoc._id));
+                  docsCreated += 1;
+                } else {
+                  throw Error('No document generated?');
+                }
+              } catch(err) {
+                console.error('#######################################');
+                console.error(`## An error occured while creating doc ${collectionDoc.document._id}`);
+                console.error(err);
+                console.error('#######################################');
+                errors += 1;
+              }
+            }
+          }
+          console.log(`Creating NRPTI collection for ${collection.displayName}`);
+          // Master, Meta and Documents for this collection are all created.
+          // Now, create a NRPTI collection and shove the docs into it!
+          let bcmiCollection = new CollectionBCMI();
+          bcmiCollection._master = new ObjectID(nrptiMine._id);
+          bcmiCollection.read = ['sysadmin', 'admin:bcmi', 'public'];
+          bcmiCollection.write = ['sysadmin', 'admin:bcmi'];
+          bcmiCollection.name = collection.displayName;
+          bcmiCollection.date = new Date();
+          bcmiCollection.project = new ObjectID(nrptiMine._id);
+          bcmiCollection.type = collection.type;
+          bcmiCollection.agency = collection.isForMEM ? 'EMPR' : collection.isForEAO ? 'EAO' : 'ENV';
+          bcmiCollection.records = allNewDocs;
+          bcmiCollection.addedBy = 'nrpti';
+          bcmiCollection.datePublished = new Date();
+          bcmiCollection.publishedBy = 'nrpti';
+          bcmiCollection.isBcmiPublished = true;
```

Comment:
Just confirming we want this auto-published?

Reply:
I assumed so, as they're published in mem-admin, but we can always leave that as an exercise for an admin if it's safer